### PR TITLE
update dependencies, add build step to lint. 

### PR DESCRIPTION
**Pull request by kgrubb**
__CHANGES:__

* update dependencies
* add build step to lint
* only run release on push to master branch

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -8,8 +8,31 @@ jobs:
...
+          sudo dpkg -i wkhtmltox_0.12.6.1-2.jammy_amd64.deb
+      - name: Build project
+        run: |
+          npm ci
+          npm run compile
+          npm run release
+      - name: Upload Build Artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: microluxe20-${{ github.sha }}
```
Comment (Web-eWorks) [inline .github/workflows/lint.yml:36]: I'd like this to have a more human-readable identifier like branch name + date, but that's a very low priority ask.

**Review by sturnclaw — DISMISSED** [review]
Looks good to me! Thanks for fixing up the build system!

File: .github/workflows/lint.yml (hunk 1 above)
Comment (kgrubb) [reply .github/workflows/lint.yml]: :+1: added the branch name for the artifact identifier. I couldn't get the date to execute in the name field unfortunately.

**Review by sturnclaw — APPROVED** [review]

File: .github/workflows/lint.yml (hunk 1 above)
Comment (Web-eWorks) [reply .github/workflows/lint.yml]: I took a look at the docs, it does appear to be non-trivial to get the date. This is good enough!

**Comment by Web-eWorks** [comment]
Interestingly enough, I appear to be unauthorized to merge PRs to the master branch, even though this PR has met the review criteria. Unsure if that's on purpose or not.

**Comment by kgrubb** [comment]
You should have permissions to approve and merge now!

**Pull request merged**